pubtools/_pulp/tasks/push/items/erratum.py

```diff
@@ -29,13 +29,17 @@ class ErratumUploadContext(UploadContext):
 class PulpErratumPushItem(PulpPushItem):
     """Handler for errata."""
 
-    # Erratums are always uploaded to this repo first.
-    # Pulp stores the erratum pkglist for (erratum_id, repo) pair. If an erratum is
-    # uploaded for an already existing ID, the pkglist is overwritten if the repo is
-    # same, else is appended. To keep the pkglist consistent, all the erratums are
-    # uploaded to the same repo instead of a random repo from the list of destinations.
-    # For more details, refer RHELDST-12782.
-    UPLOAD_REPO = "all-rpm-content"
+    MULTI_UPLOAD_CONTEXT = True
+
+    @property
+    def upload_repo(self):
+        # Split errata into different repos by year, assuming the advisory name
+        # is formatted like RHXA-YYYY
+        try:
+            return "all-erratum-content-%s" % self.pushsource_item.name[5:9]
+        except KeyError:
+            # Poorly formatted errata name, use default
```

Comment:
This seems like dead code.  Are you sure this can be reached?

If we assume that `pushsource_item.name` at least is a `str`, I don't think we can get here.

Firstly, if you index a string (or list) out of bounds, you don't get KeyError, you get IndexError.

Secondly, if you *slice* a string (or list) out of bounds, you don't get an error at all, you get an incomplete (or empty) sequence of the same type.

```
>>> x = 'abc'

>>> x[5]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: string index out of range

>>> x[1:9]
'bc'

>>> x[5:9]
''
```